### ISSUE TITLE
Better ping but slow speed (Need another better way to test speed!

### ISSUE THREAD

**Issue by Jason4V** [AdguardTeam/AdGuardVPNForAndroid#20]
Hi, loyal pro paid user for both adguard desktop and mobile here. Just purchased VPN Annual Subscription then found a big problem: (shot on browser extension)

![ag vpn](https://user-images.githubusercontent.com/385914/93598362-27e46180-f9ef-11ea-9d06-d8b63b47e0dc.png)

Please help to look into this issue! Maybe we need another way to test speed rather than ping! @ameshkov

**Comment by zzebrum** [comment]
@Jason4V sorry for the late reply! Devs are looking into it 👌

**Comment by Jason4V** [comment]
Thanks @zzebrum Well, I thought it'll be the limit of speed test method by using ping time, so, we should use another way to solve this problem.
Anyway, thanks adguard team bring so many great products and look fwd your feedback

**Comment by ameshkov** [comment]
> Maybe we need another way to test speed rather than ping! 

Well, ping indeed does not guarantee better speed. Most of the time it correlates okay, but cases like yours are still possible.

The problem is that doing a proper speed check would be too expensive - we would have to download a substantially-sized file from every endpoint (for instance, a 1-10MB file).

**Comment by Jason4V** [comment]
> The problem is that doing a proper speed check would be too expensive - we would have to download a substantially-sized file from every endpoint (for instance, a 1-10MB file).

Hi, @ameshkov The method you mentioned is accepted by many vpn clients. 10MB file for each line may be expensive, but cutomers like me have to use sth like youtube 4K video to check speed which is much more than 10MB! 

Think about it, 「customer better experience > better reputation > better sale」 or 「Cost savings, more profit」 which is the way that a startup product shall take?

**Comment by ameshkov** [comment]
It's expensive not just for us, it's expensive for the customers as well.

Providing some sort of a test page that you could use once to measure every endpoint may be feasible, replacing pings with this approach is definitely not.

**Comment by Jason4V** [comment]
> Providing some sort of a test page that you could use once to measure every endpoint may be feasible, replacing pings with this approach is definitely not.

Sure. Pings is fast, download is slow but reliable. We can provide both :)

**Comment by ameshkov** [comment]
Yep, not a bad idea, I was thinking about providing some sort of a speed test for every endpoint

**Comment by Jason4V** [comment]
> Yep, not a bad idea, I was thinking about providing some sort of a speed test for every endpoint

Cool! Looking forward this new feature.  Really appreciate @ameshkov

**Comment by artemiv4nov** [comment]
I'll close this task because we already have a speed test for every endpoint.
Also, the network speed has been enhanced a long time ago.

**Comment by Jason4V** [comment]
Thank you! @artemiv4nov
